### PR TITLE
Fedora Core 41 build support

### PR DESCRIPTION
**Pull request by m0r1k**
Hi guys!

Thanks for your work. After a whole day of suffering, I managed to run this on Fedora Core 41 (gcc version 14.2.1 20240912 (Red Hat 14.2.1-3) (GCC)). Maybe my script will be useful to someone (I found many posts on the net, about how Fedora Core users tried to compile this)

Regards,

### CODE REVIEW

File: utils/backscrub.sh [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+BACKSCRUB_BUILD_DIR=$BACKSCRUB_GIT_DIR/build/
+CPU_CORES=$(grep -c processor /proc/cpuinfo)
+CAPTURE_DEV=/dev/video1
+VIRT_DEV_ID=0
+VIRT_DEV=/dev/video$VIRT_DEV_ID
+TARGET=/usr/local/bin/backscrub
+BACK_GROUND=$BACKSCRUB_GIT_DIR/backgrounds/rotating_earth.webm
+ACTION=$1
+
+set -ue;
```
Comment (BenBE) [inline utils/backscrub.sh:23]: No `;` needed at the end of bash commands.

File: utils/backscrub.sh [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+function do_checks
+{
+    RES=`cat /etc/*lease | grep -i fedora || /bin/true`;
+    if [ ! -n "$RES" ];then
```
Comment (BenBE) [inline utils/backscrub.sh:28]: `grep -q` is your friend.

File: utils/backscrub.sh [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+VERSION=20241106140722
+
+##
+## https://github.com/floe/backscrub
+##
+## backscrub, Fedora build script
+## Roman E. Chechnev <m0r1k>
+##
+
+ROOT_DIR=~/Downloads/backscrub_patched/
```
Comment (BenBE) [inline utils/backscrub.sh:12]: This should be something the user can specify. Not everyone has that downloads folder called this way …

File: utils/backscrub.sh [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    if [ ! -r "$FNAME" ]; then
+        sudo cat <<EOF > $FNAME
+v4l2loopback
+EOF
+    fi
+}
+
+function do_modprobe
+{
+    sudo modprobe v4l2loopback;
```
Comment (BenBE) [inline utils/backscrub.sh:94]: This should check if the module hasn't been loaded before, thus avoiding annoying prompts for your root password if the module has been loaded at boot time.

File: utils/backscrub.sh [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+BACK_GROUND=$BACKSCRUB_GIT_DIR/backgrounds/rotating_earth.webm
+ACTION=$1
+
+set -ue;
+
+function do_checks
+{
+    RES=`cat /etc/*lease | grep -i fedora || /bin/true`;
+    if [ ! -n "$RES" ];then
+        echo "sorry, only Fedora supported now";
```
Comment (BenBE) [inline utils/backscrub.sh:29]: Which part of it only works on Fedora to warrant this error?

File: utils/backscrub.sh [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    do_setup;
+    do_modprobe;
+    do_start;
+elif [ "$ACTION" == "start" ]; then
+    do_modprobe;
+    do_start;
+else
+    do_usage;
+fi
+
```
Comment (BenBE) [inline utils/backscrub.sh:146]: Unnecessary blank line at EOF.

**Review by BenBE — CHANGES_REQUESTED** [review]

**Comment by m0r1k** [comment]
ahah, it's so funny, maybe you misunderstood, your support of Fedora Core 41 is 0%, forget, I have my own fork :)

File: utils/backscrub.sh (hunk 6 above)
Comment (m0r1k) [reply utils/backscrub.sh:146]: read what the rules of "good tone" are

File: utils/backscrub.sh (hunk 5 above)
Comment (m0r1k) [reply utils/backscrub.sh:29]: have you tested it somewhere else?

File: utils/backscrub.sh (hunk 4 above)
Comment (m0r1k) [reply utils/backscrub.sh:94]: it's modprobe's work to do this check, before asking for the privilege

File: utils/backscrub.sh (hunk 3 above)
Comment (m0r1k) [reply utils/backscrub.sh:12]: ahah

File: utils/backscrub.sh (hunk 2 above)
Comment (m0r1k) [reply utils/backscrub.sh:28]: fix your code for Fedora Core, that's who your friend is )

File: utils/backscrub.sh (hunk 1 above)
Comment (m0r1k) [reply utils/backscrub.sh:23]: yes, yes, ";" No need here, ";" No need there, and then in Kotlin you can’t do this:
```
var str = "Test"
  + "Test"
  + "Test"
```

**Comment by BenBE** [comment]
Nobody working on this project is being paid for the work. It's all done in our spare time, of which there is actually quite little. The comments on this PR are mostly about changes that should be fixed to have a uniform code style or things that can be improved to support a broader usecase, instead of being just you and your usecase. The build issues your script addresses are by no means unique to Fedora AFAICT, thus this script would have been a good compromise for the short-term if it wasn't for your personality.

**Pull request closed without merging**